modules/exploits/windows/scada/rockwell_factorytalk_rce.rb

```diff
@@ -0,0 +1,225 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Powershell
+  include Msf::Exploit::Remote::HttpServer
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Rockwell FactoryTalk View SE SCADA Unauthenticated Remote Code Execution',
+        'Description' => %q{
+          This module exploits a series of vulnerabilities to achieve unauthenticated remote code execution
+          on the Rockwell FactoryTalk View SE SCADA product.
+          The default configuration is exploitable by an unauthenticated attacker, and the exploit runs as
+          the IIS user on the Windows server.
+          The attack relies on the chaining of five separate vulnerabilities. The first vulnerability is an unauthenticated project copy request,
+          the second is a directory traversal, and the third is a race condition. In order to achieve full remote code execution on all
+          targets, two information leak vulnerabilities are also abused.
+          This exploit was used by the Flashback team (Pedro Ribeiro + Radek Domanski) in Pwn2Own Miami 2020 to win the EWS category.
+        },
+        'License' => MSF_LICENSE,
+        'Author' =>
+        [
+          'Pedro Ribeiro <pedrib[at]gmail.com>', # Vulnerability discovery and Metasploit module
+          'Radek Domanski <radek.domanski[at]gmail.com>' # Vulnerability discovery and Metasploit module
+        ],
+        'References' =>
+          [
+            [ 'URL', 'https://www.thezdi.com/blog/2020/7/22/chaining-5-bugs-for-code-execution-on-the-rockwell-factorytalk-hmi-at-pwn2own-miami'],
+            [ 'URL', 'PEDRO_BLOG'],
+            [ 'URL', 'RADEK_BLOG'],
+            [ 'CVE', '2020-12027'],
+            [ 'CVE', '2020-12028'],
+            [ 'CVE', '2020-12029'],
+            [ 'ZDI', '20-727'],
+            [ 'ZDI', '20-728'],
+            [ 'ZDI', '20-729'],
+            [ 'ZDI', '20-730'],
+          ],
+        'Privileged' => false,
```

Comment:
If the exploit gain `SYSTEM` privileges:

```suggestion
        'Privileged' => true,
```

Reply:
Nope, that's not correct, it runs as the IIS user

Reply:
Previously your description of this module in PR did actually say that it was getting code execution as SYSTEM, so this may have been a point of confusion. I have updated the PR description to note that it gains code execution as the IIS user.